src/claimtrie.cpp

```diff
@@ -312,57 +352,72 @@ void completeHash(uint256& partialHash, const std::string& key, std::size_t to)
             .Finalize(partialHash.begin());
 }
 
-bool recursiveCheckConsistency(CClaimTrie::const_iterator& it, int minimumHeight, std::string& failed)
+template <typename T>
+using iCbType = std::function<void(T&)>;
+
+template <typename TIterator>
+uint256 recursiveMerkleHash(TIterator& it, const iCbType<TIterator>& process, const iCbType<TIterator>& verify = {})
 {
     std::vector<uint8_t> vchToHash;
-
     const auto pos = it.key().size();
-    auto children = it.children();
-    for (auto& child : children) {
-        if (!recursiveCheckConsistency(child, minimumHeight, failed))
-            return false;
+    for (auto& child : it.children()) {
+        process(child);
```

Comment:
"process" is pretty vague. Do we have a more specific name? And this is just to allow a few lines of code reuse between validation and hash computation?

Reply:
Suggest better naming, i'll update it.